components/wmtk_components/multimesh/multimesh.cpp

```diff
@@ -0,0 +1,33 @@
+#include "multimesh.hpp"
+
+#include "MultimeshOptions.hpp"
+
+
+namespace wmtk::components {
+void multimesh(const nlohmann::json& j, io::Cache& cache)
+{
+    MultimeshOptions options = j.get<MultimeshOptions>();
+
+    std::shared_ptr<Mesh> parent = cache.read_mesh(options.parent);
+    std::shared_ptr<Mesh> child = cache.read_mesh(options.child);
+
+    if (parent->top_simplex_type() == child->top_simplex_type() &&
+        parent->capacity(parent->top_simplex_type()) ==
+            child->capacity(child->top_simplex_type())) {
+        std::vector<int64_t> parent_simplices(parent->capacity(parent->top_simplex_type()));
+        std::iota(parent_simplices.begin(), parent_simplices.end(), 0);
+
+        auto child_map =
+            multimesh::same_simplex_dimension_surjection(*parent, *child, parent_simplices);
```

Comment:
i guess for now this is just pipelining to enable obj-style mappings? we could clarify that i guess. need a better name for all these things

Reply:
yes, i have a throw for the other cases, we can add as needed